### PR TITLE
Add AquaDocs

### PR DESCRIPTION
**Pull request by adam3smith**
Requested: https://fediscience.org/@thalassa/109303603306517407

Much of the DC data is not actually in the page header & not in the default output formats, so going through the XML.

### CODE REVIEW

File: AquaDocs.js [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+	let xmlText = await requestText(xmlURL);
+	// Z.debug(xmlText)
+	var xml = (new DOMParser()).parseFromString(xmlText, "text/xml");
+	var ns = {
+		dim: 'http://www.dspace.org/xmlns/dspace/dim',
+		mets: 'http://www.loc.gov/METS/',
+		xlink: 'http://www.w3.org/TR/xlink/'
+	};
+
+	let type = ZU.xpathText(xml, '//dim:field[@element="type"]', ns);
```
[the remark this answers is not in the document]
Comment (adam3smith) [reply AquaDocs.js:112]: It's used internally in dspace, but I'm not sure how standardized it actually is. Let me find that out.

[the remark this answers is not in the document]
Comment (adam3smith) [reply AquaDocs.js]: Done

**Comment by adam3smith** [comment]
Linted successfully locally.

**Comment by adam3smith** [comment]
@AbeJellinek  -- are you still reviewing translators? If so, could you take a look?

**Comment by adam3smith** [comment]
From my point this is gtg

File: AquaDocs.js [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+	// Z.debug(xmlURL);
+	let xmlText = await requestText(xmlURL);
+	// Z.debug(xmlText)
+	let translator = Zotero.loadTranslator('import');
+	translator.setTranslator('2c05e2d1-a533-448f-aa20-e919584864cb'); // DIM
+	translator.setString(xmlText);
+	translator.setHandler('itemDone', (_obj, item) => {
+		for (let attachment of item.attachments) {
+			if (attachment.url && !attachment.url.startsWith("http")) {
+				attachment.url = "https://aquadocs.org" + item.attachments[i].url;
```
[the remark this answers is not in the document]
Comment (adam3smith) [reply AquaDocs.js]: Yes -- this was actually introduced by Abe fixing up my code which was still using a counter. This is cleaner, though

File: AquaDocs.js [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (items) {
+			await Promise.all(
+				Object.keys(items)
+					.map(url => requestDocument(url).then(scrape))
```
[the remark this answers is not in the document]
Comment (adam3smith) [reply AquaDocs.js]: Thanks for catching -- that *was* the template back when I wrote the translator (IIRC there's an open issue created by dstillman to fix all of the ones based on that version of the template)

[the remark this answers is not in the document]
Comment (dstillman) [reply AquaDocs.js]: Yup:

https://github.com/zotero/translators/issues/2980
https://github.com/zotero/translators/issues/2983

**Comment by adam3smith** [comment]
OK, had a rebase issue in between, but looks all good now @zoe-translates . Files lint correctly, but all PRs with import translators currently get stuck in CI.

**Comment by adam3smith** [comment]
@zoe-translates -- thanks, makes sense on both & tests look good. Should be ready to merge now

**Pull request merged**